### DISCUSSION TITLE
Is there a way to switch to a different buffer by name?

### DISCUSSION THREAD

**Discussion by icefoxen** [helix-editor/helix#6057] [Ideas] [answered]
Longtime vim user just playing with Helix, and one thing I miss is being able to go to a buffer with `:b th<tab>` to autocomplete from the open buffer list.  But it appears that while Helix has `buffer_next` and `buffer_prev` commands, it doesn't have just a "switch to buffer" command.  Doing `<spc> b` and having to go up and down to choose the buffer I want is nice when I have a ton of buffers, but less effective when I know that I'm messing with code in 4-5 places and know what the buffers are called.  Is this a itch anyone else wants to scratch?  Is it worth trying to make a PR for it?

**Comment by LEI** [comment, marked as answer]
You can type the name in the buffer picker (`<Space>b`), it's not autocomplete but should filter the list.

The PR #4756 adds a `buffer-goto` command accepting an index, the next step might be the name.

**Comment by icefoxen** [comment]
Ah, the filtering is *almost* as nice as autocomplete, thanks!

That PR is...  well.  For me choosing buffers by index rather than name seems almost worse than nothing at all.  Sure, multiple buffers can have the same name, but that's why you disambiguate them as `foo.txt` and `foo.txt <2>` and such like vim does.
